dsl_parser/tasks.py

```diff
@@ -34,21 +34,40 @@ def parse_dsl(dsl_location,
 def _set_plan_inputs(plan, inputs=None):
     inputs = inputs if inputs else {}
     # Verify inputs satisfied
+    missing_inputs = []
     for input_name, input_def in plan['inputs'].iteritems():
         if input_name not in inputs:
             if 'default' in input_def and input_def['default'] is not None:
                 inputs[input_name] = input_def['default']
             else:
-                raise exceptions.MissingRequiredInputError(
-                    "Required input '{0}' was not specified - expected "
-                    "inputs: {1}".format(input_name, plan['inputs'].keys()))
+                missing_inputs.append(input_name)
+
+    if missing_inputs:
+        if len(missing_inputs) == 1:
```

Comment:
I don't see why we need to treat a single input differently. Let's just state "missing inputs" even if there is only one missing input.

Thanks